### ISSUE TITLE
Update sync to run quarterly instead of annually

### ISSUE THREAD

**Issue by daissatou2** [OpenFn/primero-ona-dashboard#17]
## Background, context, and business value
This sync was scheduled to happen annually with a static cursor. The client has requested that we update the workflow to now run quarterly. 

## The specific request, in as few words as possible
We want to sync cases that have been created/updated since the last sync. 

Update the [Get cases from Primero](https://github.com/OpenFn/primero-ona-dashboard/blob/staging/jobs/getCasesFromPrimero.js) job: 
1. instead of `createdAt` in lines 26 and 54, we should be fetching data where `last_updated_at` is greater than or equal to the cursor date
2. At the end of the job, we should set `dateCursor` equal to the time the last sync started. 

## state.json

Job 0: 
{"inboxUrl":"ADD URL"}

Job 1:
```json
{
  "configuration": ["SEE LAST PASS: 'Primero Cambodia V2 Demo (api_user)'"],
  "data": {"NA" },
  "cursor": "2023-01-01T00:00:00.000Z"
}
```

Job 2:
```json
{
  "configuration": ["SEE LAST PASS: 'ONA Test Database'"],
  "data": {"NA" },
  "cursor": "2023-01-01T00:00:00.000Z"
}
```

Job 3:
{"inboxUrl":"ADD URL"}

## adaptor
Job 0: `http`
Job 1: `primero`
Job 2: `postgreSQL` 
Job 3: `http`

## trigger

Trigger to be used on production -example: Job 0 is `cron` (quarterly), the rest are`flow`jobs

## expression.js
Job 0: https://github.com/OpenFn/primero-ona-dashboard/blob/staging/jobs/0-initiate-sync.js
Job 1: https://github.com/OpenFn/primero-ona-dashboard/blob/staging/jobs/getCasesFromPrimero.js
Job 2: https://github.com/OpenFn/primero-ona-dashboard/blob/staging/jobs/upsertCasesToONA.js
Job 3: https://github.com/OpenFn/primero-ona-dashboard/blob/staging/jobs/3-request-next-batch.js'

## output
To test this, use the scenarios below. Test with case that has "case_id_display" = "1db2a39" and currently has a `last_updated_date` of `2023-03-28T06:55:31.494Z`. You can use this case to test both GET requests. 

1. set the manual cursor to a date _before_ `2023-03-28T06:55:31.494Z`, clear state, and run the job - case with "case_id_display" = "1db2a39" _should_ have been returned from job 1. Run the job again (without clearing state) - case with "case_id_display" = "1db2a39" _SHOULD NOT_ have been returned from job 1. 
2. set the manual cursor to _equal_ `2023-03-28T06:55:31.494Z`, clear state, and run the job. Case with "case_id_display" = "1db2a39" _should_ have been returned from job 1. 
3. set the manual cursor to sometime _after_ `2023-03-28T06:55:31.494Z`, clear state, and run the job. Case with "case_id_display" = "1db2a39" _SHOULD NOT_ have been returned from job 1.

**Comment by mtuchi** [comment]
For 2. See this example from primero thailand 
```
let lastRunDateTime = filteredCases
    .map(c => c.last_updated_at)
    .sort((a, b) => new Date(b.date) - new Date(a.date))[0];

  lastRunDateTime =
    new Date(lastRunDateTime) > new Date()
      ? lastRunDateTime
      : new Date().toISOString();
```

**Comment by daissatou2** [comment]
Hey @mtuchi we're getting zero cases returned for both GET requests with `2023-03-26T06:55:31.494Z`as the cursor. See this run [log](https://www.openfn.org/projects/primero-ona-cambodia-io-staging/runs/064c0551-ad37-78f6-b8f3-65754da04d19). 

I know this should return at least one case: case with "case_id_display" = "1db2a39". This is because that case has `"last_updated_at": = "2023-03-28T06:55:31.494Z"`, `"age" = 14` and `"type_of_case" = "children_undergoing_reintegration_55427"`.  See this GET cases [response](https://www.openfn.org/projects/primero-ona-cambodia-io-staging/runs/064bfa4d-96e3-7119-81bf-5ce5da34de19) for just that case. 

I did some troubleshooting. When I set the cursor equal to `2023-03-28T06:55:31.494Z`, _I do_ get that case back. So we know the "equal to" scenario is working. But the "greater than" scenario is not. 

Below are some examples of how a similar requirement was implemented on other projects. @aleksa-krolls I'm a bit confused by these examples below. It looks like we're actually fetching cases where the last updated date _equals_ the cursor and is similar to what Mtuchi has implemented for this issue. But at least according to the comments in the job and from my memory, the requirement was cases created/updated since the last sync. Am I missing something here?
- [UNICEF Cambodia line 22](https://github.com/OpenFn/unicef-cambodia/blob/e608eb386889a85b9b2296accb16d8c008e43568/jobs/V2/f1-j1-getPrimeroCasesV2.js#L22. )
- [UNICEF Thailand line 22 ](https://github.com/OpenFn/primero-thailand/blob/e3e55e3d07476e71e80f266d94bfd57051ea2ee4/jobs/child-shield/getCasesFromPrimero.js#L22)

**Comment by aleksa-krolls** [comment]
@daissatou2  My understanding is that if you append `..` to the query date, Primero should return cases created on or _after_ this date. E.g.,`"last_updated_at": "2023-03-28T06:55:31.494Z.."`  (<- note the trailing `..` after the time stamp). 

@mtuchi please investigate to confirm.

**Comment by mtuchi** [comment]
@daissatou2 i have added the date range for the `manualCursor` as well. Before you will get the exact date cases because the dateCursor was setup to be exact for a `manualCursor`.  I have merged this PR #19 , Give it another try and let me know 
if your still facing some challenges
